src/common/version.ts

```diff
@@ -49,15 +53,42 @@ export class Versions {
         return Versions.getValue(Constants.versionTempSensor, "1.0") as string;
     }
 
+    public static updateSystemModuleImageVersion(templateJson: any, moduleName: string, versionMap: Map<string, string>) {
+        if (templateJson) {
+            const sysModuleImage =
+                templateJson.modulesContent.$edgeAgent["properties.desired"].systemModules[moduleName].settings.image;
+            templateJson.modulesContent.$edgeAgent["properties.desired"].systemModules[moduleName].settings.image =
+                Versions.getNewImageVersionJson(sysModuleImage, versionMap);
+        }
+    }
+
+    public static edgeHubVersion(): string {
+        return Versions.getDefaultEdgeRuntimeVersion();
+    }
+
     private static edgeAgentVersion(): string {
-        return Versions.getValue(Constants.versionEdgeAgent) as string;
+        return Versions.getDefaultEdgeRuntimeVersion();
     }
 
-    private static edgeHubVersion(): string {
-        return Versions.getValue(Constants.versionEdgeHub, "1.0") as string;
+    private static getDefaultEdgeRuntimeVersion(): string {
+        return Versions.getValue(Constants.versionDefaultEdgeRuntime, "1.0") as string;
+    }
+
+    private static getNewImageVersionJson(input: string, versionMap: Map<string, string>): string {
```

Comment:
maybe you should define a type for the imageName, since it looks like you're expecting a certain form of string.

Reply:
something like:
```ts
type  fooType =`$${string}:${string}`
```

also based on the type definitions, `input` cannot be undefined, so `if (input) {' is both vague and redundant. If you're checking that it's not an empty string, maybe define it as `input !== ''`